### ISSUE TITLE
Duplicate icon definitions causing module to crash Nuxt with "Failed to fetch dynamically imported module" error

### ISSUE THREAD

**Issue by wagerfield** [swisnl/nuxt-lucide-icons#1]
There are a number of icons with the same _name_ with different _capitalisation_ causing this module to crash Nuxt.

Here you can see the `Grid2x2` and `Grid3x3` icons imported from the official `lucide-vue-next` package with different casing for the `x` character eg. `Grid2x2` and `Grid2X2`

<img width="311" alt="Screenshot 2023-11-08 at 13 52 57" src="https://github.com/swisnl/nuxt-lucide-icons/assets/412747/34aad79c-3b41-446b-b012-7159a20e6345">

These duplicate icons are causing the following warnings to be printed in the terminal.

> Note: I'm using "Icon" as the `namePrefix`

<img width="1171" alt="Screenshot 2023-11-08 at 13 22 25" src="https://github.com/swisnl/nuxt-lucide-icons/assets/412747/f4e594ed-7197-45e0-9881-dcf97572db35">

This causes Nuxt to crash upon page load with the error: `Failed to fetch dynamically imported module`

The clashing icons are:

| Icon              | Duplicate         |
| ----------------- | ----------------- |
| ArrowDownAZ | ArrowDownAz |
| ArrowDownZA | ArrowDownZa |
| ArrowUpAZ   | ArrowUpAz   |
| ArrowUpZA   | ArrowUpZa   |
| Axis3D      | Axis3d      |
| FileAxis3D  | FileAxis3d  |
| Grid2x2     | Grid2X2     |
| Grid3x3     | Grid3X3     |
| Move3D      | Move3d      |
| Rotate3D    | Rotate3d    |
| Scale3D     | Scale3d     |

Not sure what the best remedy is to this problem, but one idea is to `filter` out the duplicate variations [as your are doing here](https://github.com/swisnl/nuxt-lucide-icons/blob/main/src/module.ts#L32-L33) for the icons starting with `Lucide` and ending in `Icon`

This is how I would do it with a RegEx:

```ts
Object.keys(icons)
  .filter((icon) => icon !== 'default')
  .filter((icon) => !icon.startsWith('Lucide'))
  .filter((icon) => !icon.endsWith('Icon'))
  .filter((icon) => !/(Az|Za|3d|2X2|3X3)$/.test(icon)) // <-- filter out all icons ending in "Az", "Za", "3d", "2X2" and "3X3"
```

It's obviously not a bulletproof solution...but I tested it locally and it resolves the issue.

An alternative would be to have an `exclude` list that you manually add the full icon names that you want to filter out:

```ts
const exclude = [
  "ArrowDownAz",
  "ArrowDownZa",
  "ArrowUpAz",
  "ArrowUpZa",
  "Axis3d",
  "FileAxis3d",
  "Grid2X2",
  "Grid3X3",
  "Move3d",
  "Rotate3d",
  "Scale3d",
]

Object.keys(icons)
  .filter((icon) => icon !== 'default')
  .filter((icon) => !icon.startsWith('Lucide'))
  .filter((icon) => !icon.endsWith('Icon'))
  .filter((icon) => !exclude.includes(icon))
```

**Comment by JaZo** [comment]
I've chosen to specify a priority as suggested by the warning, as overwriting is what's currently happening and hopefully will be foolproof (see #2). Let me know what you tink of this solution!

**Comment by JaZo** [comment]
Fix released in [v1.0.3](https://github.com/swisnl/nuxt-lucide-icons/releases/tag/v1.0.3)!
